lib/ansible/modules/cloud/amazon/elasticache_parameter_group.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = """
+---
+module: elasticache
+short_description: Manage cache security groups in Amazon Elasticache.
+description:
+  - Manage cache security groups in Amazon Elasticache.
+  - Returns information about the specified cache cluster.
+version_added: "2.3"
+author: "Sloane Hertel (@s-hertel)"
+options:
+  group_family:
+    description:
+      - The name of the cache parameter group family that the cache parameter group can be used with.
+    choices: ['memcached1.4', 'redis2.6', 'redis2.8', 'redis3.2']
+    required: yes
+    type: string
+  name:
+    description:
+     - A user-specified name for the cache parameter group.
+     required: yes
+     type: string
+  description:
+    description:
+      - A user-specified description for the cache parameter group.
+  state:
+    description:
+      - Idempotent actions that will create/modify, destroy, or reset a cache parameter group as needed.
+    choices: ['present', 'absent', 'reset']
+    required: true
+  values:
+    description:
+      - A user-specified list of parameters to reset or modify for the cache parameter group.
+    required: no
+    type: list
+    default: None
+"""
+
+EXAMPLES = """
+# Note: None of these examples set aws_access_key, aws_secret_key, or region.
+# It is assumed that their matching environment variables are set.
+---
+- hosts: localhost
+  connection: local
+  tasks:
+    - name: 'Create a test parameter group'
+      elasticache_parameter_group:
+        name: 'test-param-group'
+        group_family: 'redis3.2'
+        description: 'This is a cache parameter group'
+        state: 'present'
+    - name: 'Modify a test parameter group'
+      elasticache_parameter_group:
+        name: 'test-param-group'
+        values:
+          - ['activerehashing', 'yes']
```

Comment:
Instead of taking a list of `[name, value]`pairs, it'd be nice to accept a map like:

```
values:
  activerehashing: yes
  something: no
```